perllib/FixMyStreet/DB/ResultSet/Problem.pm

```diff
@@ -173,7 +173,11 @@ sub around_map {
             longitude => { '>=', $p{min_lon}, '<', $p{max_lon} },
     };
 
-    my $age_field = $c->stash->{sort_key} =~ /updated/ ? 'lastupdate' : 'confirmed';
+    # this is a { -desc => order } hash so flatten to array
```

Comment:
It's only that if it's `desc`, if it was ascending then it's only a string: lastupdate/confirmed/comment_count.
(and if it's comment_count you won't want to be using that for a time comparison :)